### PR TITLE
Add description annotation

### PR DESCRIPTION
**Pull request by danilbykov**
Can we add annotation `@description` to fill descriptions of fields in `Schema`s?

### CODE REVIEW

**Comment by adamw** [comment]
Thanks!

Sure, that would be a welcome addition to the way schema is derived. One thing to consider - maybe we should instead have a `@schema` annotation with the metadata that can be specified (`description`, `format` and `deprecated`)? This would also leave the door open to adding more meta-data, if there would be a need. In fact, in the case of products, the field name could be customised as well (currently this is only possible through a global strategy `sttp.tapir.generic.Configuration`.

Another thing that would be nice to have is a mention in the docs on how the annotation can be used.

**Comment by danilbykov** [comment]
I have tried to add such annotation. We should not require to specify all parameters so this annotation must have default values, for example,
```scala
class tschema(val desc: String = "", val deprecated = false)
```
or
```scala
class tschema(val desc: Option[String] = None, val deprecated: Option[Boolean] = None)
```

But there is open [issue in scala compiler](https://github.com/scala/bug/issues/9612) for annotations with default arguments.

This issue probably affects magnolia somehow because I even can not compile following code
```scala
@tschema(deprecated = true)
case class I(field: Int)
```
While this version is compiled fine
```scala
@tschema(desc = "asdf")
case class I(field: Int)
```

To overcome this issue we may use with several different annotations: `description`, `format`, `name` and so on. Users probably will not use all of them simultaneously.

**Comment by adamw** [comment]
Ah, I was not aware of this limitation. Thanks for checking!

Then yes, makes sense to have separate annotations for description, format, deprecated and fieldName. Would you be able to add these or do we keep the PR in its current scope?

**Comment by danilbykov** [comment]
I added four annotations: `@description`, `@format`, `@deprecated` and `@name`.

> In fact, in the case of products, the field name could be customised as well

Do you mean tuples? Currently components of tuple in schema have name like `_1`, `_2` and so on. Annotations can not be applied to components of tuple so annotation `@name` can not customize these `_1`, `_2`, ....

I will update doc soon.

**Comment by adamw** [comment]
> Do you mean tuples?

No, I meant case classes :) As it is now, thanks a lot for the changes!

File: core/src/test/scala/sttp/tapir/generic/SchemaGenericTest.scala [hunk 1]
```diff
@@ -182,6 +182,27 @@ class SchemaGenericTest extends AnyFlatSpec with Matchers {
...
 
+  it should "add description to schema from annotations" in {
+    val schema = implicitly[Schema[I]]
+    schema shouldBe Schema[I](
+      SProduct(
+        SObjectInfo("sttp.tapir.generic.I"),
+        List(
+          (FieldName("int"), intSchema.description("some int field").format("int32")),
+          (FieldName("noDesc"), longSchema),
+          (FieldName("alternativeBooleanName"), implicitly[Schema[Option[Boolean]]].description("another optional boolean flag")),
```
[the remark this answers is not in the document]
Comment (danilbykov) [reply core/src/test/scala/sttp/tapir/generic/SchemaGenericTest.scala:193]: What about `@specificationName` or `@specName`?

[the remark this answers is not in the document]
Comment (adamw) [reply core/src/test/scala/sttp/tapir/generic/SchemaGenericTest.scala:193]: Hm I don't think that "specification" applies here. What would be the specification to which it refers?

The name *is* the "low-level" or "raw" name in the serialized form of an object.

I think I would go with `@serializedName` if you think that would be good enough. Naming is hard ;)

**Comment by adamw** [comment]
Released in [0.17.0-M2](https://github.com/softwaremill/tapir/releases/tag/v0.17.0-M2), thanks!

In the end I went with `@encodedName`, since we have a `Codec` with `.encode`

**Pull request merged**